crates/subspace-service/src/lib.rs

```diff
@@ -502,7 +504,13 @@ where
     let executor = sc_service::new_wasm_executor(&config.executor);
     let domains_executor = sc_service::new_wasm_executor(&config.executor);
 
-    let backend = sc_service::new_db_backend(config.db_config())?;
+    let confirmation_depth_k = extract_confirmation_depth(config.chain_spec.as_ref())
+        .expect("Confirmation depth is always present in Genesis chain spec; qed");
```

Comment:
I'd prefer to return an error here, this function is already fallible, so not difficult to do